src/plugins/SysML/SysMLExporter/BlockDefinitionDiagramExporter.js

```diff
@@ -0,0 +1,377 @@
+/*globals define*/
+/*jshint node:true, browser:true*/
+
+/**
+ * Author: Vishwesh Nath
+ * Created on: November 31, 2015
+ */
+
+define(['ejs',
+        'plugin/SysMLExporter/SysMLExporter/Templates/Templates',
+        './SysMLExporterConstants'], function (ejs, TEMPLATES, CONSTANTS) {
+
+    'use strict';
+
+    var BlockDefinitionDiagramExporter = function () {
+    };
+	
+	var webgmeEclipseMapping ={
+		'Block' : 'Class'
+	};
+	
+	var templateMapping = {
+		'Class' : 'Block.ejs'
+	};
+
+    BlockDefinitionDiagramExporter.prototype.addComponent = function (nodeObj) {
+
+        var self = this,
+            core = self.core,
+            gmeID = core.getPath(nodeObj),
+            baseClass = self.getMetaType(nodeObj),
+            type = core.getAttribute(baseClass, 'name'),
+            name = core.getAttribute(nodeObj, 'name'),
+            xPos = core.getRegistry(nodeObj, 'position').x,
+            yPos = core.getRegistry(nodeObj, 'position').y,
+            element,
+            parentPath = core.getPath(core.getParent(nodeObj)),
+            diagramKey = parentPath + "+" + core.getAttribute(nodeObj.parent, 'name');
+
+        if (self.isMetaTypeOf(baseClass, self.META.Block)) {
+
+            self.idLUT[gmeID] = {id: self.modelID};
+            self.reverseIdLUT[self.modelID] = gmeID;
+
+            element = {
+                name: name,
+                id: self.modelID,
+                x: xPos,
+                y: yPos,
+                type: webgmeEclipseMapping[type]
+            };
+            //actor = ejs.render(TEMPLATES['actor.uml.ejs'], {actorId: self.modelID, x: xPos, y: yPos});
+
+            if (!self.blockdefinitionDiagrams.hasOwnProperty(diagramKey)) {
+                self.blockdefinitionDiagrams[diagramKey] = {};
+            }
+            if (!self.blockdefinitionDiagrams[diagramKey].hasOwnProperty('elements')) {
+                self.blockdefinitionDiagrams[diagramKey].elements = [];
+            }
+            self.blockdefinitionDiagrams[diagramKey].elements.push(element);
+            self.modelID += 1;
+        }
+    };
+
+    BlockDefinitionDiagramExporter.prototype.addConnection = function (nodeObj, callback) {
+
+        var self = this,
+            core = self.core,
+            parentPath = core.getPath(core.getParent(nodeObj)),
+            diagramKey = parentPath + "+" + core.getAttribute(nodeObj.parent, 'name'),
+            src = core.getPointerPath(nodeObj, "src"),
+            dst = core.getPointerPath(nodeObj, "dst"),
+            type = core.getAttribute(self.getMetaType(nodeObj), 'name'),
+            counter = 2,
+            error = '',
+            pushUseCaseLink,
+            afterSrcLoaded,
+            afterDstLoaded,
+            srcMetaType,
+            dstMetaType,
+            srcX,
+            srcY,
+            dstX,
+            dstY,
+            srcName,
+            dstName,
+            srcId,
+            dstId;
+
+        pushUseCaseLink = function (err, shouldPush) {
```

Comment:
shouldPush seems redundant. It's false only when there is an error but at line 98 there's already a check for error.